UKHomeOffice/cop-ui
# COP-2827+COP-2829: Cases search functionality and history panel

**AC**
- replicate the behaviour from COP-v1 (cop-private-workflow-tasklist) for the cases page.

**Description**
- This branch adds the case search functionality - a user can search a business key or keyword and search results are displayed in left hand panel. 
- When the user selects a case link the details panel appears. So far, the only sections populated are the heading and the Case History section. 

**To test:** 
- Pull branch and run development server against the Dev environment (change proxy in package json). This is because the workflow-service latest build is still not working and you need to have the latest version of the API to get this to work. 
- Navigate to http://localhost:3000/cases and input a search. Results will appear on the left hand panel if there are hits. Select a case and the details panel will show with the Case history showing the process instances.
- "Show Details" form snapshot on Case History not implemented yet.
- npm run test to see tests for CasePage, CaseResults panel, Case History.

**NB:** 
- Cases API does not always bring back the correct "Number of cases found" - This is the same on V1.
- If a search has +20 hits the API paginates results. However, this doesn't seem to always work correctly, e.g. search keyword: abigail - it will say there are 40 results, over 2 pages, but "page 1" only has 6 results and the next has 14.
